### ISSUE TITLE
[REVIEW]: KMC_Lattice v2.0: An Object-Oriented C++ Library for Custom Kinetic Monte Carlo Simulations

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#1168]
**Submitting author:** @MikeHeiber (<a href="http://orcid.org/0000-0002-1567-5663">Michael Heiber</a>)
**Repository:** <a href="https://github.com/MikeHeiber/KMC_Lattice" target ="_blank">https://github.com/MikeHeiber/KMC_Lattice</a>
**Version:** v2.0.1
**Editor:** @katyhuff
**Reviewer:** @myousefi2016, @mdoucet
**Archive:** <a href="https://doi.org/10.5281/zenodo.2552685" target="_blank">10.5281/zenodo.2552685</a>

## Status

[![status](http://joss.theoj.org/papers/db7676d3ffd001b9a981fbbec513394c/status.svg)](http://joss.theoj.org/papers/db7676d3ffd001b9a981fbbec513394c)

Status badge code:

```
HTML: <a href="http://joss.theoj.org/papers/db7676d3ffd001b9a981fbbec513394c"><img src="http://joss.theoj.org/papers/db7676d3ffd001b9a981fbbec513394c/status.svg"></a>
Markdown: [![status](http://joss.theoj.org/papers/db7676d3ffd001b9a981fbbec513394c/status.svg)](http://joss.theoj.org/papers/db7676d3ffd001b9a981fbbec513394c)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/MikeHeiber/KMC_Lattice" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@myousefi2016 & @mdoucet, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.theoj.org/about#reviewer_guidelines. Any questions/concerns please let @katyhuff know.

✨ **Please try and complete your review in the next two weeks** ✨ 

### Conflict of interest

- [x] As the reviewer I confirm that I have read the [JOSS conflict of interest policy](https://github.com/openjournals/joss/blob/master/COI.md) and that there are no conflicts of interest for me to review this work.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/MikeHeiber/KMC_Lattice">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Version:** v2.0.1
- [x] **Authorship:** Has the submitting author (@MikeHeiber) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the function of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Authors:** Does the `paper.md` file include a list of authors with their affiliations?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?

### Conflict of interest

- [x] As the reviewer I confirm that I have read the [JOSS conflict of interest policy](https://github.com/openjournals/joss/blob/master/COI.md) and that there are no conflicts of interest for me to review this work.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/MikeHeiber/KMC_Lattice">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Version:** v2.0.1
- [x] **Authorship:** Has the submitting author (@MikeHeiber) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the function of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Authors:** Does the `paper.md` file include a list of authors with their affiliations?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @myousefi2016, it looks like you're currently assigned as the reviewer for this paper :tada:.

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
PDF failed to compile for issue #1168 with the following error: 

   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100    15    0    15    0     0    180      0 --:--:-- --:--:-- --:--:--   182
Error reading bibliography ./paper.bib (line 47, column 1):
unexpected "t"
expecting space, ",", white space or "}"
Error running filter pandoc-citeproc:
Filter returned error status 1
Looks like we failed to compile the PDF

**Comment by MikeHeiber** [comment]
For the latest draft of the paper, please see the joss-edits branch.

**Comment by MikeHeiber** [comment]
@whedon generate pdf from branch joss-edits

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch joss-edits. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01168/joss.01168/10.21105.joss.01168.pdf)

**Comment by myousefi2016** [comment]
@katyhuff I'm done with the review and everything looks good!

**Comment by MikeHeiber** [comment]
For full transparency, @myousefi2016 identified a bug ([Issue #62](https://github.com/MikeHeiber/KMC_Lattice/issues/62)) in KMC_Lattice through testing out the [KMC_Lattice_example](https://github.com/MikeHeiber/KMC_Lattice_example/) example program.  The issue is now resolved in the KMC_Lattice joss-edits branch and will be merged into master with all other corrections/improvements once the review phase is completed.

**Comment by katyhuff** [comment]
Excellent. Thanks for a thorough review @myousefi2016 . It looks like @mdoucet is making good progress as well. I'll have a chance to look this submission over myself sometime this weekend.

**Comment by mdoucet** [comment]
I'm done with my review. The code is well organized and the examples are pretty good. Installing and running the code was easy. There's enough references to dig deeper and understand the problem. Overall everything looks good!

**Comment by katyhuff** [comment]
@whedon generate pdf from branch joss-edits

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch joss-edits. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01168/joss.01168/10.21105.joss.01168.pdf)

**Comment by katyhuff** [comment]
Thanks @myousefi2016 and @mdoucet for your reviews !

@MikeHeiber thank you for a strong submission and for engaging actively in the review process! I have looked over the paper, double checked all the DOI links, and have conducted a high level review of the code itself. Everything looks ship-shape to me.
Once you have your code and paper all ready in the release version (I assume v2.0.0?), it'll be good to double check the paper, review any lingering details in your code/readme/etc., and then make an archive of the reviewed software in Zenodo/figshare/other service. When you update this thread with the DOI of the archive, I'll move forward with accepting the submission! Until then, now is your moment for final touchups!

**Comment by MikeHeiber** [comment]
Thank you @katyhuff @myousefi2016 and @mdoucet for reviewing my software tool and making this a smooth process.  I look forward to doing this again soon once my Excimontec KMC software tool is complete.

KMC_Lattice v2.0.0 is now released and the paper looks ready to go.  v2.0.0 is archived on Zenodo with the following DOI:
10.5281/zenodo.2547139

**Comment by MikeHeiber** [comment]
@katyhuff Is there anything else I can do to help things move forward?

**Comment by katyhuff** [comment]
Sorry, I just took a break from this over the weekend. Let's move forward. Thanks for the reminder.

**Comment by katyhuff** [comment]
I'm going to double check a bunch of things while in meetings today. Should be able to complete publication later this afternoon.

**Comment by katyhuff** [comment]
@whedon set 10.5281/zenodo.2547139 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.2547139" target="_blank">10.5281/zenodo.2547139</a> is the archive.

**Comment by katyhuff** [comment]
@whedon set v2.0.0 as version

**Comment by katyhuff** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by katyhuff** [comment]
@MikeHeiber Thank you for again for your strong submission and for prompt responses to review comments. I approve acceptance of this paper!

@mdoucet @myousefi2016 Thank you again for your reviews. We couldn't do it without you!

@arfon this paper is ready to accept -- over to you!

**Comment by danielskatz** [comment]
I'm checking on something before I finalize this...

**Comment by MikeHeiber** [comment]
@danielskatz Looking over the final proof, I found a small typo error in the manuscript.  In one sentence, the word "the" appears twice in a row.  How should I go about correcting this?

**Comment by danielskatz** [comment]
Just fix it in the .md file, and let me know

**Comment by MikeHeiber** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01168/joss.01168/10.21105.joss.01168.pdf)

**Comment by MikeHeiber** [comment]
@katyhuff @danielskatz The paper is now corrected, and a new release has been issued.  Please change the release to v2.0.1 with the DOI: 10.5281/zenodo.2552685

**Comment by danielskatz** [comment]
@whedon set v2.0.1 as version

**Comment by whedon** [comment]
OK. v2.0.1 is the version.

**Comment by danielskatz** [comment]
@whedon set 10.5281/zenodo.2552685 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.2552685" target="_blank">10.5281/zenodo.2552685</a> is the archive.

**Comment by danielskatz** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/458

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/458, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by whedon** [comment]
```Reference check summary:

OK DOIs

- http://doi.org/10.1016/0021-9991(75)90060-1 is OK
- http://doi.org/10.1088/1751-8121/aa95c7 is OK
- http://doi.org/10.1016/0021-9991(76)90041-3 is OK
- http://doi.org/10.1146/annurev.physchem.58.032806.104637 is OK
- http://doi.org/10.1016/j.cpc.2014.04.003 is OK
- http://doi.org/10.1016/j.cpc.2014.04.017 is OK
- http://doi.org/10.1016/j.pmatsci.2017.09.003 is OK
- http://doi.org/10.1088/0953-8984/27/1/013001 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by danielskatz** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/459
1. Wait a couple of minutes to verify that the paper DOI resolves [https://doi.org/10.21105/joss.01168](https://doi.org/10.21105/joss.01168)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? notify your editorial technical team...

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](http://joss.theoj.org/papers/10.21105/joss.01168/status.svg)](https://doi.org/10.21105/joss.01168)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.01168">
  <img src="http://joss.theoj.org/papers/10.21105/joss.01168/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: http://joss.theoj.org/papers/10.21105/joss.01168/status.svg
   :target: https://doi.org/10.21105/joss.01168
```
This is how it will look in your documentation:

[![DOI](http://joss.theoj.org/papers/10.21105/joss.01168/status.svg)](https://doi.org/10.21105/joss.01168)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: http://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://www.flipcause.com/secure/cause_pdetails/Mjk3Nzk=
